messaging/src/test/java/org/axonframework/commandhandling/gateway/IntervalRetrySchedulerTest.java

```diff
@@ -0,0 +1,91 @@
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.GenericCommandMessage;
+import org.axonframework.common.jdbc.JdbcException;
+import org.junit.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import static org.junit.Assert.*;
+
+/**
+ * Test the {@link IntervalRetryScheduler}.
+ *
+ * @author Bert Laverman
+ */
+public class IntervalRetrySchedulerTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private static final int RETRY_INTERVAL = 100;
+    private static final int MAX_RETRIES = 5;
+
+    private IntervalRetryScheduler retryScheduler;
+
+    @Before
+    public void setup() {
+        retryScheduler = IntervalRetryScheduler
+                .builder()
+                .retryExecutor(new ScheduledThreadPoolExecutor(1))
+                .retryInterval(RETRY_INTERVAL)
+                .maxRetryCount(MAX_RETRIES)
+                .build();
+    }
+
+    /**
+     * Schedule a retry, faking that we had a certain nr of (transient) failures.
+     *
+     * @param retryScheduler the scheduler to use.
+     * @param nrOfFailures   the number of (transient) failures .
+     * @return the nr of milliseconds delay actually achieved.
+     */
+    static long doScheduleRetry(RetryScheduler retryScheduler, int nrOfFailures) {
```

Comment:
I can imagine we move this to a `RetrySchedulerTestUtil` of some sorts later on. For now though, I am fine with this approach.